### PR TITLE
[otelecho] Use SpanStatusFromHTTPStatusCodeAndSpanKind on the middleware

### PR DESCRIPTION
**Pull request by sgirones**
Fix the `otelecho` middleware and follow the OpenTelemetry specification.

For servers, 4XX status codes must not be treated as errors. OpenTelemetry's `semconv` package provides the `SpanStatusFromHTTPStatusCodeAndSpanKind` so the status code is treated different depending on whether the `SpanKind` is `server` or not

### CODE REVIEW

**Comment by pellared** [comment]
Hello @sgirones 👋 
Are you able to sign the CLA so that we can accept your contribution?

**Comment by sgirones** [comment]
@pellared done!

**Review by pellared — APPROVED** [review]

**Comment by pellared** [comment]
I think it is worth adding an entry to `CHANGELOG.md`

**Comment by sgirones** [comment]
Thanks for the hand-holding @pellared. I added a changelog line.

Let me know if I there's anything missing I can help with.

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,10 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ## [Unreleased]
 
+### Fixed
+
+- Fix the `echo` middleware by using `SpanKind.SERVER` when deciding the `SpanStatus`. This makes `4xx` response codes to not be an error anymore.
```
Comment (MrAlias) [inline CHANGELOG.md:13]:
```suggestion
- Fix the `echo` middleware by using `SpanKind.SERVER` when deciding the `SpanStatus`.
  This makes `4xx` response codes to not be an error anymore. (#1848)
```

**Review by MrAlias — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**